app/src/main/java/com/luc4n3x/levyra/feature/recognition/MusicRecognitionController.kt

```
@@ -0,0 +1,125 @@
+            activeJob.also { activeJob = null }
+        }
+        job?.cancel()
+    }
+
+    private suspend fun runRecognition() {
+        _state.value = RecognitionState.Listening
+        val captured = try {
+            audioCapture.capture(captureDurationMs)
+        } catch (error: CancellationException) {
+            throw error
+        } catch (error: MicrophonePermissionDeniedException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)